java-checks/src/main/java/org/sonar/java/checks/spring/StatusCodesOnResponseCheck.java

```diff
@@ -0,0 +1,150 @@
+
+import java.util.List;
+import java.util.stream.Stream;
+import org.sonar.check.Rule;
+import org.sonar.java.model.ExpressionUtils;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S6863")
+public class StatusCodesOnResponseCheck extends IssuableSubscriptionVisitor {
+
+  public static final String RESPONSE_ENTITY = "org.springframework.http.ResponseEntity";
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return List.of(Tree.Kind.CLASS);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    ClassTree classTree = (ClassTree) tree;
+    if (!isClassController(classTree)) {
+      return;
+    }
+
+    MethodInvocationVisitor methodInvocationVisitor = new MethodInvocationVisitor();
+    classTree.accept(methodInvocationVisitor);
+  }
+
+  private class MethodInvocationVisitor extends BaseTreeVisitor {
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree methodInvocationTree) {
+
+      MethodMatchers statusMethodMatchers = MethodMatchers.create()
+        .ofTypes(RESPONSE_ENTITY)
+        .names("status")
+        .addParametersMatcher("org.springframework.http.HttpStatus")
+        .build();
+
+      if (statusMethodMatchers.matches(methodInvocationTree)) {
+        checkTryCatchForStatus(methodInvocationTree);
+      }
+
+      MethodMatchers okMethodMatchersWithParam = MethodMatchers.create()
+        .ofTypes(RESPONSE_ENTITY)
+        .names("ok")
+        .withAnyParameters()
+        .build();
+
+      if (okMethodMatchersWithParam.matches(methodInvocationTree)) {
+        Tree catchParent = checkCatchWithErrorStatus(methodInvocationTree, false);
+
+        if (catchParent == null) {
+          checkTryWithOkStatus(methodInvocationTree, true);
+        }
+      }
+
+      MethodMatchers errorMethodMatchersWithParam = MethodMatchers.create()
+        .ofTypes(RESPONSE_ENTITY)
+        .names("badRequest", "notFound")
+        .addWithoutParametersMatcher()
+        .build();
+
+      if (errorMethodMatchersWithParam.matches(methodInvocationTree)) {
+        Tree catchParent = checkCatchWithErrorStatus(methodInvocationTree, true);
+
+        if (catchParent == null) {
+          checkTryWithOkStatus(methodInvocationTree, false);
+        }
+      }
+
+      super.visitMethodInvocation(methodInvocationTree);
+    }
+
+    private void checkTryCatchForStatus(MethodInvocationTree methodInvocationTree) {
+      Tree catchParent = ExpressionUtils.getParentOfType(methodInvocationTree, Tree.Kind.CATCH);
+      boolean isError = methodInvocationTree.arguments().stream()
+        .map(MemberSelectExpressionTree.class::cast)
```

Comment:
Is it possible to statically import the single name `NOT_FOUND` and break the cast here?